src/historywork/RunCommandWork.cpp

```diff
@@ -40,15 +40,7 @@ RunCommandWork::onRun()
 
             std::weak_ptr<RunCommandWork> weak(
                 std::static_pointer_cast<RunCommandWork>(shared_from_this()));
-            exit->async_wait([weak](asio::error_code const& ec) {
-                auto self = weak.lock();
-                if (self)
-                {
-                    self->mEc = ec;
-                    self->mDone = true;
-                    self->wakeUp();
-                }
-            });
+            exit->async_wait(wakeSelfUpCallback());
```

Comment:
This seems to be a different change and shows that the abstractions are not exactly right: we're not using any "waiting timer" in this situation, yet we call a function that talks about timers in its name...
We *could* add the concept of error code to "wakeUp" but it doesn't seem to fit well from a generic point of view, so I would keep `RunCommand` untouched.

Reply:
Agree I think it's easier to reason about this without RunCommandWork